dozer/cogs/starboard.py

```diff
@@ -0,0 +1,366 @@
+    elif len(msg.content) < ((2*MAX_EMBED) - len(view_link)):
+        e.add_field(name="Content:", value=msg.content[0:MAX_EMBED], inline=False)
+        e.add_field(name="More:", value=msg.content[MAX_EMBED:(2*MAX_EMBED) - len(view_link)] + view_link, inline=False)
+    else:
+        e.add_field(name="Content:", value=msg.content[0:MAX_EMBED], inline=False)
+        e.add_field(name="More:", value=msg.content[MAX_EMBED:2*MAX_EMBED], inline=False)
+        e.add_field(name="Link:", value=view_link)
+
+    if len(msg.attachments) > 1:
+        e.add_field(name="Attachments:", value="\n".join([f"[{a.filename}]({a.url})" for a in msg.attachments]))
+    elif len(msg.attachments) == 1:
+        if msg.attachments[0].width is not None and msg.attachments[0].filename[-4:] not in VIDEO_FORMATS:
+            e.set_image(url=msg.attachments[0].url)
+        else:
+            e.add_field(name="Attachment:", value=f"[{msg.attachments[0].filename}]({msg.attachments[0].url})")
+
+    e.set_footer(text=f"{reaction_count} reactions")
+    e.timestamp = msg.created_at
+    return e
+
+
+class Starboard(Cog):
+    """Cog to post specific 'Hall of Fame' messages in a specific channel"""
+
+    def __init__(self, bot):
+        super().__init__(bot)
+        self.config_cache = {}
+
+    def make_config_embed(self, ctx, title, config):
+        """Makes a config embed."""
+        e = discord.Embed(title=title, color=discord.Color.gold())
+        e.add_field(name="Starboard Channel", value=self.bot.get_channel(config.channel_id).mention)
+        e.add_field(name="Starboard Emoji", value=config.star_emoji)
+        e.add_field(name="Cancel Emoji", value=config.cancel_emoji)
+        e.add_field(name="Threshold", value=config.threshold)
+        e.set_footer(text=f"For more information, try {ctx.prefix}help starboard")
+        return e
+
+    async def send_to_starboard(self, config, message, reaction_count):
+        """GIven a message which may or may not exist, send it to the starboard"""
+        starboard_channel = message.guild.get_channel(config.channel_id)
+        if starboard_channel is None:
+            return
+
+        db_msgs = await StarboardMessage.get_by(message_id=message.id)
+        if len(db_msgs) == 0:
+            sent_msg = await starboard_channel.send(embed=make_starboard_embed(message, reaction_count))
+            db_msg = StarboardMessage(message.id, message.channel.id, sent_msg.id, message.author.id)
+            await db_msg.update_or_add()
+            await message.add_reaction(config.star_emoji)
+        else:
+            try:
+                sent_msg = await self.bot.get_channel(config.channel_id).fetch_message(db_msgs[0].starboard_message_id)
+            except discord.errors.NotFound:
+                # Uh oh! Starboard message was deleted. Let's try and delete it
+                DOZER_LOGGER.warning(f"Cannot find Starboard Message {db_msgs[0].starboard_message_id} to update")
```

Comment:
people deleting a starred message is probably a common enough occurrence that it would probably make more sense to set this under debug than warning, but that's my take

Reply:
Debugs don't get logged by default, warnings do. I think this is worth logging in case something else breaks here (though I could be convinced for a info instead of warning)